packages/ckeditor5-find-and-replace/src/findandreplaceediting.ts

```diff
@@ -80,15 +80,33 @@ function onDocumentChange(
 				results.remove( markerName );
 			}
 
-			writer.removeMarker( markerName );
+			// This condition is checked in several tests anyway and seems to be ignored by instabul.
+			// istanbul ignore next -- @preserve
+			if ( model.markers.has( markerName ) ) {
+				writer.removeMarker( markerName );
+			}
```

Comment:
After the research I can say this code is unnecessary. I pushed a branch with some added logs to better show what is going on here: `ck/15680-with-logs`.

Just a few lines above https://github.com/ckeditor/ckeditor5/blob/00fdb2ccfb9912a5d1379650f1c458a25e6048e1/packages/ckeditor5-find-and-replace/src/findandreplaceediting.ts#L82-L84 we update the find and replace state by removing the deleted results. This triggers the collection listener in [https://github.com/ckeditor/ckeditor5/blob/00fdb2ccfb9912a5d1379650f1c458a25e6048e1/packages/ckeditor5-find-and-replace/src/findandreplacestate.ts#L78.](https://github.com/ckeditor/ckeditor5/blob/00fdb2ccfb9912a5d1379650f1c458a25e6048e1/packages/ckeditor5-find-and-replace/src/findandreplacestate.ts#L78.) There the collection has its own logic removing the markers: [https://github.com/ckeditor/ckeditor5/blob/00fdb2ccfb9912a5d1379650f1c458a25e6048e1/packages/ckeditor5-find-and-replace/src/findandreplacestate.ts#L90-L93.](https://github.com/ckeditor/ckeditor5/blob/00fdb2ccfb9912a5d1379650f1c458a25e6048e1/packages/ckeditor5-find-and-replace/src/findandreplacestate.ts#L90-L93.) Remembering that this code is executed before the control is returned to `onDocumentChange()` callback in [https://github.com/ckeditor/ckeditor5/blob/00fdb2ccfb9912a5d1379650f1c458a25e6048e1/packages/ckeditor5-find-and-replace/src/findandreplaceediting.ts#L88-L91,](https://github.com/ckeditor/ckeditor5/blob/00fdb2ccfb9912a5d1379650f1c458a25e6048e1/packages/ckeditor5-find-and-replace/src/findandreplaceediting.ts#L88-L91,) the markers are already gone and there's nothing to remove.

Hence, AFAICS this whole block [https://github.com/ckeditor/ckeditor5/blob/00fdb2ccfb9912a5d1379650f1c458a25e6048e1/packages/ckeditor5-find-and-replace/src/findandreplaceediting.ts#L78-L93](https://github.com/ckeditor/ckeditor5/blob/00fdb2ccfb9912a5d1379650f1c458a25e6048e1/packages/ckeditor5-find-and-replace/src/findandreplaceediting.ts#L78-L93) doesn't have to be inside a `model.change()` call - we can only remove the results.

Reply:
Removed. Thx!